metals/src/main/scala/scala/meta/internal/metals/codeactions/BracelessBracefulSwitchCodeAction.scala

```diff
@@ -0,0 +1,558 @@
+            ).toSeq
+          case termTry: Term.Try =>
+            Seq(
+              createCodeActionForPotentialBlockHolder(
+                termTry.expr,
+                path,
+                document,
+                termTry.expr,
+                "try expression"
+              ),
+              createCodeActionForCatchP(
+                termTry,
+                path,
+                document,
+                "catch expression"
+              ),
+              termTry.finallyp.flatMap(finallyp =>
+                createCodeActionForPotentialBlockHolder(
+                  finallyp,
+                  path,
+                  document,
+                  finallyp,
+                  "finally expression"
+                )
+              )
+            ).flatten
+          case caseTree: Case =>
+            createCodeActionForCasesListWrapper(caseTree, path).toSeq
+          case termIf: Term.If =>
+            Seq(
+              createCodeActionForPotentialBlockHolder(
+                termIf.thenp,
+                path,
+                document,
+                termIf.thenp,
+                "then expression"
+              ),
+              createCodeActionForPotentialBlockHolder(
+                termIf.elsep,
+                path,
+                document,
+                termIf.elsep,
+                "else expression"
+              )
+            ).flatten
+          case termFor: Term.For =>
+            createCodeActionForPotentialBlockHolder(
+              termFor,
+              path,
+              document,
+              termFor.body,
+              "for expression"
+            ).toSeq
+          case termForYield: Term.ForYield =>
+            createCodeActionForPotentialBlockHolder(
+              termForYield,
+              path,
+              document,
+              termForYield.body,
+              "yield expression"
+            ).toSeq
+          case _: Term.Match => None.toSeq
+          case termWhile: Term.While =>
+            createCodeActionForPotentialBlockHolder(
+              termWhile,
+              path,
+              document,
+              termWhile.body,
+              "while expression"
+            ).toSeq
+          case _: Defn.GivenAlias => None.toSeq
+          //          case template: Template =>
+          //            createCodeActionForTemplateHolder(
+          //              template,
+          //              path,
+          //              document,
+          //              template,
+          //              "template"
+          //            ).toSeq
+          case termBlock: Term.Block =>
+            createCodeActionForPotentialBlockHolder(
+              termBlock,
+              path,
+              document,
+              termBlock,
+              "block"
+            ).toSeq
+        }
+      }
+    result.toSeq.flatten
+  }
+
+  private def hasBraces(template: Template): Boolean = {
+    util
+      .Try(template.stats.maxBy(_.pos.end).pos.end)
```

Comment:
Could we just check text(template.pos.end) == '}'?

Reply:
No. This creates a bug. Let’s say you have a braceless Template with a braced method at the end. The last token would be `}` but the Template is braceless.

Reply:
Actually, I think the last char in that case is `\n` from what I checked:

```
import scala.meta._

val code = """|object A extends B:
              |  def b = {
              |
              |  }
              |
              |""".stripMargin
implicit val dl = scala.meta.dialects.Scala3

val parsed = code.parse[Source].get

val tmpl = parsed.stats.head.asInstanceOf[Defn.Object].templ

code(tmpl.pos.end)
```

Reply:
Told you! it’s a mine field! No one else should touch this PR! haha

Reply:
So maybe we should check if the last is `}` and if that doesn't belong to the last statement?

Reply:
What I’ve written does exactly that!